papermill/abs.py

```diff
@@ -49,35 +54,32 @@ def read(self, url):
         """Read storage at a given url"""
         params = self._split_url(url)
         output_stream = io.BytesIO()
-        block_blob_service = self._block_blob_service(
-            account_name=params["account"], sas_token=params["sas_token"]
-        )
+        blob_service_client = self._blob_service_client(params["account"], params["sas_token"])
+        blob_client = blob_service_client.get_blob_client(
 
-        block_blob_service.get_blob_to_stream(
-            container_name=params["container"], blob_name=params["blob"], stream=output_stream
-        )
+            container=params['container'],
+            blob=params['blob']
 
+        )
+        blob_client.download_blob().readinto(output_stream)
         output_stream.seek(0)
         return [line.decode("utf-8") for line in output_stream]
 
     def listdir(self, url):
         """Returns a list of the files under the specified path"""
         params = self._split_url(url)
-
-        block_blob_service = self._block_blob_service(
-            account_name=params["account"], sas_token=params["sas_token"]
-        )
-        blobs = block_blob_service.list_blobs(params["container"])
-        return blobs
+        blob_service_client = self._blob_service_client(params["account"], params["sas_token"])
+        container_client = blob_service_client.get_container_client(params["container"])
+        return list(container_client.list_blobs(name_starts_with=params["blob"]))
```

Comment:
@MSeal It looks like the pipeline doesn't like converting the iterator to a list. In the unit tests it says `Mock object is not iterable`. I'm not sure how it works, I assumed that in the unit test specifying the `returnvalue` would make `listdir` return that value but it appears not. Any ideas how I can fix this?

Reply:
So what's happeneing is that during the test we replace the response from the azure library with a Mock object (a proxy that can return anything). By default it returns another Mock object for each call made to it, but you can set the `return_value` property on a mock or patch call to tell it what to return. This should be happening already on lines 28 and 37 of `test_abs.py`.  I can't run the tests from where I am atm, but to debug fruther you could print the object before we call `list` and it should list the call path made to the mock object. this should help you traceback the problem to a call that doesn't have a `return_value` set.

Reply:
I did a bit of refactoring on the unit tests, now it mocks all the methods for the BlobServiceClient, and mocks also the BlobClient and ContainerClient because those classes are where the methods have moved to.